### PR TITLE
Update CI infrastructure for v1.10 release

### PR DESCRIPTION
**Pull request by christarazi**
- test: Avoid using image digests in CI

### CODE REVIEW

**Comment by christarazi** [comment]
test-me-please

File: test/helpers/cons.go [hunk 1]
```diff
@@ -187,9 +187,9 @@ const (
 
 	// CiliumStableHelmChartVersion should be the chart version that points
 	// to the v1.X branch
-	CiliumStableHelmChartVersion = "1.9-dev"
-	CiliumStableVersion          = "v1.9"
-	CiliumLatestHelmChartVersion = "1.9.90"
+	CiliumStableHelmChartVersion = "1.10-dev"
+	CiliumStableVersion          = "v1.10"
+	CiliumLatestHelmChartVersion = "1.10.90"
```
Comment (joestringer) [inline test/helpers/cons.go]: These changes should not be backported to v1.10, as it represents the target version to start upgrade tests from.

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -92,13 +92,17 @@ var (
 	defaultHelmOptions = map[string]string{
 		"image.repository":              "k8s1:5000/cilium/cilium-dev",
 		"image.tag":                     "latest",
+		"image.useDigest":               "false",
```
Comment (joestringer) [inline test/helpers/kubectl.go:95]: This and below I think should be backported to v1.10 from what I understand.

**Review by joestringer — APPROVED** [review]

File: test/helpers/cons.go (hunk 1 above)
Comment (christarazi) [reply test/helpers/cons.go]: Hmm right, I went back and forth in my head whether to create two separate PRs, but then thought for some reason that both commits need to be backported. Good catch. I'll handle the backport myself and pull it manually into https://github.com/cilium/cilium/pull/15919.

File: test/helpers/kubectl.go (hunk 2 above)
Comment (aanm) [reply test/helpers/kubectl.go:95]: Correct

**Review by tklauser — APPROVED** [review]

**Comment by christarazi** [comment]
Dropped the commit which bumped the stable Helm releases to v1.10 as we don't have the v1.10-dev chart ready yet. A new PR will include it once we have that chart.

**Comment by christarazi** [comment]
test-me-please

**Pull request merged**